src/internal/connector/onedrive/collections.go

```diff
@@ -165,6 +166,39 @@ func (c *Collections) UpdateCollections(ctx context.Context, driveID string, ite
 	return nil
 }
 
+func shouldSkipDrive(ctx context.Context, service graph.Servicer, drivePath path.Path, m folderMatcher) bool {
+	shouldSkip := !includePath(ctx, m, drivePath)
+	if shouldSkip {
+		return true
+	}
+
+	return restrictedPath(ctx, service, drivePath)
+}
+
+func restrictedPath(ctx context.Context, service graph.Servicer, drivePath path.Path) bool {
+	folders := drivePath.Folders()
+	if len(folders) < 2 {
+		return false
+	}
+
+	driveObj, err := service.
+		Client().
+		DrivesById(folders[1]).
+		Get(ctx,
+			optionsSingleDrive([]string{"id", "name"}),
+		)
```

Comment:
Nit, func params should be inlined, or all multilined.
```suggestion
		Get(ctx, optionsSingleDrive([]string{"id", "name"}))
```